Server/server.js

```diff
@@ -1,72 +1,170 @@
+//   let product = req.body.product;
+//   let image_name = req.body.image_name;
+//   let color = req.body.color;
+//   let url = req.body.url;
+//   let alt = req.body.alt;
+
+//   Image.find({product: productParam}, function(err, result) {
+//     if (result) {
+//       Image.updateOne({product: productParam}, req.body, function(err, result) {
+//         if (err) {
+//           throw err;
+//         } else {
+//           res.send(result);
+//         }
+//       });
+//     } else {
+//       Image.create({product, image_name, color, url, alt}, function(err, result) {
+//         if (err) {
+//         throw err;
+//         } else {
+//           res.send(result);
+//         }
+//       })
+//     }
+//   });
+// })
+
+// //delete request: deletes a  data
+// app.delete('/products/:product/', function (req, res) {
+//   let productParam = req.params.product;
+
+//   Image.deleteOne({product: productParam}, function(err, result) {
+//     if (err) {
+//       throw err;
+//     } else {
+//       res.send('Got a DELETE request at /products/:product/');
+//     }
+//   });
+// })
+
+////////////////////////////////////////////////////////////////
+//cassandra
+
+const cassandra = require('cassandra-driver');
+const client = new cassandra.Client({contactPoints: ['127.0.0.1'], localDataCenter: 'datacenter1', keyspace: 'sdc'});
+client.connect((err, result) => {
+  console.log('index: cassandra connected');
 });
 
+const getAllByProduct = 'SELECT * FROM sdc.product WHERE product = ? ALLOW FILTERING';
+const getLastProduct = 'SELECT * FROM sdc.product_by_id LIMIT 1 ALLOW FILTERING';
+const postByProduct = 'INSERT INTO sdc.product (product, image_name, color, url, category, picture_length, alt) VALUES = ? ALLOW FILTERING';
+
 app.get('/products/:product/', function(req, res) {
-  var productParam = req.params.product;
-  var colorParam = req.params.color;
-  Image.find({
-      product: productParam
-}, function(err, result) {
+  let productParam = Number(req.params.product);
+  console.log('server');
+  client.execute(getAllByProduct, [productParam], {hints : ['int']}, (err, result) => {
     if (err) {
-    throw err;
-  } else {
-    res.setHeader('Access-Control-Allow-Origin', '*');
-    res.send(result);
-  }
+      res.status(404).send(err);
+    } else {
+      res.status(200).send(result.rows);
+    }
   });
-
 });
 
-
-////////////////////////////////////////////////////
-//new addition
-
 //post request: creates a new data
 app.post('/products/', function(req, res) {
-  var product = req.body.product;
-  var imageName = req.body.imageName;
-  var color = req.body.color;
-  var url = req.body.url;
-  var alt = req.body.alt;
-
-  Image.create({product, imageName, color, url, alt}, function(err, result) {
+  let image_name = req.body.image_name;
+  let color = req.body.color;
+  let url = req.body.url;
+  let category = req.body.category;
+  let picture_length = req.body.picture_length;
+  let alt = req.body.alt;
+  let product;

```

Comment:
could use destructuring to make this a little cleaner: 

`let {image_name, color, url, category, picture_length, alt} = req.body`

Reply:
that is definitely much cleaner. I will use that!